static/app/components/forms/controls/selectControl.tsx

```diff
@@ -189,14 +184,6 @@ function SelectControl<OptionType extends GeneralSelectValue = GeneralSelectValu
 
   const defaultStyles = useMemo<StylesConfig>(
     () => ({
-      container: provided => ({
-        ...provided,
-        ...(isCompact && {
-          display: 'flex',
-          flexDirection: 'column',
-          maxHeight: maxMenuHeight,
-        }),
-      }),
```

Comment:
Removing all the conditional `CompactSelect` styles (activated when `isCompact` is true) from `SelectControl`, now that the two are completely separate.